### PR TITLE
Returned previous behaviour.Duration mustn't be required field

### PR DESCRIPTION
**Pull request by savvinsergey**

### CODE REVIEW

File: src/gantt.ts [hunk 1]
```diff
@@ -671,104 +671,112 @@ module powerbi.extensibility.visual {
...
-                            } while (task.daysOffList.length && datesDiff - DaysInAWeekend > DaysInAWeek);
+                const selectionId: powerbi.extensibility.ISelectionId = selectionBuider.createSelectionId();
+                const extraInformation: ExtraInformation[] = [];
+                const resource: string = (values.Resource && values.Resource[index] as string) || "";
+                const startDate: Date = (values.StartDate
+                    && Gantt.isValidDate(values.StartDate[index] as Date) && values.StartDate[index] as Date)
+                    || new Date(Date.now());
+
+                if (values.ExtraInformation && Object.keys(values.ExtraInformation).length) {
+                    for (const key of Object.keys(values.ExtraInformation)) {
```
Comment (ignatvilesov) [inline src/gantt.ts]: Looks like this construction computes properties every step of this cycle. I think it'd be good to use `Object.keys(values.ExtraInformation).forEach` instead.

File: src/gantt.ts [hunk 2]
```diff
@@ -671,104 +671,112 @@ module powerbi.extensibility.visual {
...
-                                }
-                            } while (task.daysOffList.length && datesDiff - DaysInAWeekend > DaysInAWeek);
+                const selectionId: powerbi.extensibility.ISelectionId = selectionBuider.createSelectionId();
+                const extraInformation: ExtraInformation[] = [];
+                const resource: string = (values.Resource && values.Resource[index] as string) || "";
+                const startDate: Date = (values.StartDate
+                    && Gantt.isValidDate(values.StartDate[index] as Date) && values.StartDate[index] as Date)
+                    || new Date(Date.now());
+
+                if (values.ExtraInformation && Object.keys(values.ExtraInformation).length) {
```
Comment (ignatvilesov) [inline src/gantt.ts]: Please keep Object.keys(values.ExtraInformation) to a local variable in order not to compute keys twice.

File: test/_references.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@
 /// <reference path="../node_modules/@types/jasmine-jquery/index.d.ts" />
 
 // Power BI API
-/// <reference path="../.api/v1.6.0/PowerBI-visuals.d.ts" />
+/// <reference path="../.api/v1.7.5/PowerBI-visuals.d.ts" />
```
Comment (ignatvilesov) [inline test/_references.ts]: 1.7.5? This should be 1.7.0 instead

File: test/visualTest.ts [hunk 4]
```diff
@@ -117,6 +117,16 @@ module powerbi.extensibility.visual.test {
                 });
             });
 
+            it("Tasks Elements is present in dom, if only available task name", (done) => {
```
Comment (ignatvilesov) [inline test/visualTest.ts]: `Task Elements are presented in DOM if and only if task name is available (specified)`

File: src/gantt.ts (hunk 2 above)
Comment (savvinsergey) [reply src/gantt.ts]: Done

File: src/gantt.ts (hunk 1 above)
Comment (savvinsergey) [reply src/gantt.ts]: Done

File: test/_references.ts (hunk 3 above)
Comment (savvinsergey) [reply test/_references.ts]: Yes, sure. Done

File: test/visualTest.ts (hunk 4 above)
Comment (savvinsergey) [reply test/visualTest.ts]: Done

**Pull request merged**